src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,91 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int CAPACITY_MULTIPLIER = 2;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getBucketIndex(key);
+        for (Node<K, V> node = table[index]; node != null; node = node.next) {
+            if ((node.key == key) || (key != null && key.equals(node.key))) {
+                node.value = value;
+                return;
+            }
+        }
+        if (++size > (int) (table.length * DEFAULT_LOAD_FACTOR)) {
+            resize();
+        }
+        index = getBucketIndex(key);
+        addMyNode(key, value, index);
     }
 
     @Override
     public V getValue(K key) {
+        int index = getBucketIndex(key);
+        for (Node<K, V> node = table[index]; node != null; node = node.next) {
+            if (node.key == key || (key != null && key.equals(node.key))) {
+                return node.value;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    public void addMyNode(K key, V value, int index) {
+        Node<K, V> firstNode = table[index];
+        table[index] = new Node<>(key, value, firstNode);
+    }
+
+    private void transfer(Node<K, V>[] newTable) {
+        for (Node<K, V> currentNode : table) {
+            for (Node<K, V> node = currentNode; node != null;) {
+                Node<K, V> nextNode = node.next;
+                int index = getNewBucketIndex(node.key, newTable.length);
+                node.next = newTable[index];
+                newTable[index] = node;
+                node = nextNode;
+            }
+        }
+    }
+
+    private int getNewBucketIndex(K key, int newTableLength) {
+        int hash = (key == null) ? 0 : key.hashCode();
+        return (hash & 0x7fffffff) % newTableLength;
+    }
+
+    private int getBucketIndex(K key) {
+        int hash = (key == null) ? 0 : key.hashCode();
```

Comment:
What if we change this implementation to?
```suggestion
        return (key == null) ? 0 : Math.abs(key.hashCode() % table.length);
```